common/app/templates/inlineJS/nonBlocking/detectAdblock.scala.js

```diff
@@ -0,0 +1,24 @@
+try {
+    ((document, window) => {
+        var ad = document.createElement('div');
+        ad.style.position = 'absolute';
+        ad.style.left = '-9999px';
+        ad.style.height = '10px';
+        ad.innerHTML = '&nbsp;';
+        ad.setAttribute('class', 'ad_unit');
+
+        window.requestAnimationFrame(() => {
+            document.body.appendChild(ad);
+
+            window.requestAnimationFrame(() => {
```

Comment:
Actually, you are not waiting for the browser to apply styles to `ad_unit`. You are waiting for the browser to send a DOM mutation event of some kind to a listening extension, and for that extension to mutate the DOM on behalf of the user. There are actually quite a few opportunities for race conditions, and this is one reason why we're not currently able to track AdblockPlus in Internet Explorer.